### PR TITLE
refactor(DW): improve dataset's constructors and related tests

### PR DESCRIPTION
**Pull request by pddpd**
### PR includes

- [ ] fixed #0
- [x] add / modify test cases
- [ ] documents, demos

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -22,22 +22,22 @@ module.exports = {
...
+    'no-shadow': 0,
+    'no-console': 0,
+    'arrow-body-style': 0,
+    'no-useless-constructor': 0,
     'no-sparse-arrays': 0,
     'no-inner-declarations': 0,
-    'no-use-before-define': 'off',
-    '@typescript-eslint/no-use-before-define': ['error'],
+    'no-use-before-define': 0,
+    '@typescript-eslint/no-use-before-define': 2,
```
Comment (neoddish) [inline .eslintrc.js:40]: Why numbers are better than semantic text?

Comment (pddpd) [reply .eslintrc.js:40]: I just want to merge them. Only one style needs to exist. @neoddish

**Review by neoddish — APPROVED** [review]
LGTM

**Pull request merged**